lib/snaplist_bloc.dart

```diff
@@ -143,26 +145,41 @@ class SnapListBloc {
 
   _calculateScrollProgress(double currentPosition) {
     final distance = (_startPosition - currentPosition).abs();
-    return ((distance * 100) /
-            sizeProvider(_centerItemPosition, _createBuilderData()).width)
-        .clamp(0.0, 100.0);
+    double sizeFactor = isVertical ?
```

Comment:
I'd moved the `sizeProvider(_centerItemPosition, _createBuilderData())` to a different line, and then read the `width` or `height` from it's result. I believe it's simpler that way.